pnet-idp-client-saml2/src/main/java/at/porscheinformatik/idp/saml2/Saml2ResponseProcessor.java

```diff
@@ -123,5 +125,16 @@ public Saml2AuthenticationToken getToken()
             return token;
         }
 
+        public Optional<String> getRequestedRelayState()
+        {
+            String relayState = token.getAuthenticationRequest().getRelayState();
+            return Optional.ofNullable(relayState);
+        }
+
+        public Optional<String> getResponseRelayState()
+        {
+            HttpRequestContext details = HttpRequestContext.fromToken(token);
+            return Saml2Utils.getRelayState(details.getRequest());
```

Comment:
Why though? We didn't change the signature of this method? It still expects `HttpServletRequest`
```java
public static Optional<String> getRelayState(HttpServletRequest request)
```
right?

Reply:
We changed `Saml2Utils.getRelayState` to only return the part of the RelayState, that was added by the user. It is possible that we get a empty Optional, instead of a UUID, if the user provided no additional data in the RelayState.

But this method is used to compare the RelayState from the AuthnRequest, with the RelayState that the Identity Provider used in his response. They must be exactly the same, otherwise the comparison fails, and the login is not possible.

For this reason, we can not use the Saml2Utils anymore. Instead we must load the raw RelayState from the Request.

Reply:
Ah okay, so the goal here is not to use `Saml2Utils.getRelayState`.